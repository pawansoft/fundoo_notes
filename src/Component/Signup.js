import React, { Component } from 'react';
import {
    TextInput,
    View,
    TouchableOpacity,
    Text,
    Image
} from 'react-native';
import RegisterStyle from '../Style/Register';

export default class Signup extends Component {

    constructor(props){
        super(props);
        this.state = {
            fnameValid: '',
            fname: '',
            lnameValid: '',
            lname: '',
            userNameValid: '',
            userName: '',
            passwordValid: '',
            password: '',
            passMatch: '',
            confirm: '',

        }
    }

    validateFName = async() =>{
        const regex = /[A-Z][a-z]{3,}/
        if(regex.test(this.state.fname) == false){
            await this.setState({
                fnameValid : 'invalid',
            })
        }
        else{
            await this.setState({
                fnameValid : ''
            })
        } 
    }

    validateLName = async() => {
        const regex = /[A-Z][a-z]{3,}/
        if(regex.test(this.state.lname) == false){
            await this.setState({
                lnameValid : 'invalid',
            })
        }
        else{
            await this.setState({
                lnameValid : ''
            })
        } 
    }

    validateUserName = async() => {
        const regex = /^[0-9a-zA-Z]+([._+-][0-9a-zA-Z]+)*[@][0-9A-Za-z]+([.][a-zA-Z]{2,4})*$/
        if(regex.test(this.state.userName) == false ){
            await this.setState({
                userNameValid : 'Invalid Email'
            })
        }else{
            await this.setState({
                userNameValid : ''
            })
        }
    }

    validatePassword = async() => {
        const regex = /^(?=.*[A-Z])(?=.*[a-z])(?=.*[0-9])(?=.*[*.!@#$%^&(){}:'<>,.>/~`_+=|].).{8,}$/
        if(regex.test(this.state.password) == false ){
            await this.setState({
                passwordValid : 'Not Strong'
            })
        }else{
            await this.setState({
                userNameValid : ''
            })
        }
    }

    comparePassword = async() => {
        if(this.state.password == this.state.confirm && this.state.passwordValid != ''){
            await this.setState({
                passMatch: 'Wrong Password'
            })           
        }
        else{
            await this.setState({
                passMatch: ''
            })
        }
    }

    handleFName = async(fname) => {
        await this.setState({
            fname : fname
        })
    }

    handleLName = async(lname) => {
        await this.setState({
            lname : lname
        })
    }

    handleUserName = async(userName) => {
        await this.setState({
            userName : userName
        })
    }

    handlePassword = async(password) => {
        await this.setState({
            password : password
        })
    }

    handleConfirm = async(confirm) => {
        await this.setState({
            confirm : confirm
        })
    }

    render() {
        return (
            <View>

                <View>
                    <Image source={require('../assets/logo.png')} style = {{marginLeft : '20%'}}/>
                </View>

                <View>
                    <Text style = {RegisterStyle.signin_text}>
                        Sign Up
                    </Text>
                </View>

                <View style={{ flexDirection: 'row' }}>
                    <View style={RegisterStyle.row_text_input}>
                    <TextInput
                        placeholder={'First Name'}
                        value = {this.state.fname} 
                        onChangeText = {this.handleFName}
                        onEndEditing = {this.validateFName}/>
                         <Text style = {RegisterStyle.error_first_name}>{this.state.fnameValid}</Text>
                    </View>
                   
                    <View style={RegisterStyle.row_text_input}>
                    <TextInput
                        value = {this.state.lname}
                        onChangeText = {this.handleLName}
                        placeholder={'Last Name'}
                        onEndEditing = {this.validateLName} />
                        <Text style = {RegisterStyle.error_first_name}>{this.state.lnameValid}</Text>
                    </View>
                </View>

                <View>
                    <TextInput
                        style={RegisterStyle.text_input}
                        value = {this.state.userName}
                        onChangeText = {this.handleUserName}
                        placeholder={'Username'}
                        onEndEditing = {this.validateUserName} />
                        <Text style = {RegisterStyle.error_first_name}>{this.state.userNameValid}</Text>
                </View>
                
                <View style={{ flexDirection: 'row' }}>
                    <View style = {RegisterStyle.row_text_input}>
                    <TextInput
                        secureTextEntry={true}
                        value = {this.state.password}
                        onChangeText = {this.handlePassword}
                        onEndEditing = {this.validatePassword}
                        placeholder={'Password'} />
                        <Text style = {RegisterStyle.passcode_error}>{this.state.passwordValid}</Text>
                    </View>
                    <View style={RegisterStyle.row_text_input}>
                    <TextInput
                        secureTextEntry = {true}
                        value = {this.state.confirm}
                        onChangeText = {this.handleConfirm}
                        onEndEditing = {this.comparePassword}
                        placeholder={'Confirm'} />
                        <Text style = {RegisterStyle.passcode_error}>{this.state.passMatch}</Text>
                </View >
                </View>

                <View style={{ flexDirection: 'row' }}>
                    
                    <TouchableOpacity style = {RegisterStyle.touchable_opacity_style}>
                        <Text style ={RegisterStyle.button_Text}> 
                            Sign in instead
                        </Text>
                    </TouchableOpacity>
                    
                    <TouchableOpacity style = {RegisterStyle.button}>
                        <Text style ={RegisterStyle.button_Text}>
                            Next
                        </Text>
                    </TouchableOpacity>
                </View>
            </View>
        )
    }
}
