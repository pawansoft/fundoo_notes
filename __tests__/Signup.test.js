import React from 'react';
import Adapter from 'enzyme-adapter-react-16'
import {shallow, configure} from 'enzyme'
import Signup from '../src/Component/Signup';
import MockAsyncStorage from 'mock-async-storage';

const mockImpl = new MockAsyncStorage();
jest.mock('@react-native-async-storage/async-storage', () => mockImpl);

jest.mock('react-native-fetch-blob', () => {
    return {
      DocumentDir: () => {},
      polyfill: () => {},
    }
  });

jest.mock('react-native-localization', () => class RNLocalization {
    language = 'en-US'
  
    constructor (props) {
      this.props = props
      this.setLanguage(this.language)
    }
  
    setLanguage (interfaceLanguage) {
      this.language = interfaceLanguage
      if (this.props[interfaceLanguage]) {
        var localizedStrings = this.props[this.language]
        for (var key in localizedStrings) {
          if (localizedStrings.hasOwnProperty(key)) this[key] = localizedStrings[key]
        }
      }
    }
})

configure({adapter: new Adapter()})

describe('Signup Test', () => {
    it('ProvideRender_WhenMatchToSnapshot_TestShouldPass', () => {
        const component = shallow(<Signup/>)
        expect(component).toMatchSnapshot()
    })
})
describe('Test Signup Text Field', () => {
  
    it('ProvideTextFieldValue_WhenAdded_ShouldStoreAtState', () =>{
        const component = shallow(<Signup/>)
        component.instance().handleFName('Pawan');
        expect(component.instance().state.fname).toBe('Pawan')
    })

    it('ProvideLastName_WhenPassedToTextInput_ShouldStoredAtState', () => {
        const component = shallow(<Signup/>)
        component.instance().handleLName('Kumar');
        expect(component.instance().state.lname).toBe('Kumar')
    })

    it('ProvideUserName_WhenPassedToTextInput_ShouldStoredAtState', () => {
        const component = shallow(<Signup/>)
        component.instance().handleUserName('pk.soft29@gmail.com');
        expect(component.instance().state.userName).toBe('pk.soft29@gmail.com')
    })

    it('ProvidePassword_WhenPassedToTextInput_ShouldStoredAtState', () => {
        const component = shallow(<Signup/>)
        component.instance().handlePassword('pksoft29@gmail');
        expect(component.instance().state.password).toBe('pksoft29@gmail')
    })

    it('ProvideConfirmPassword_WhenPassedToTextInput_ShouldStoredAtState', () => {
        const component = shallow(<Signup/>)
        component.instance().handleConfirm('pksoft29@gmail');
        expect(component.instance().state.confirm).toBe('pksoft29@gmail')
    })
})
describe('Test Regex validation',() => {
    it('ProvideFirstName_WhenInvalid_ShouldGenerateError', () => {
        const component = shallow(<Signup/>) 
        component.instance(). handleFName('pawan');
        component.instance().validateFName();
        expect(component.instance().state.fnameValid).toBe('invalid')
    })

    it('ProvideFirstName_WhenValid_ErrorMessageshouldREmoved', () => {
        const component = shallow(<Signup/>) 
        component.instance(). handleFName('pawan');
        component.instance().validateFName();
        expect(component.instance().state.fnameValid).toBe('invalid')
        component.instance().handleFName('Pawan');
        component.instance().validateFName();
        expect(component.instance().state.fnameValid) .toBe ('')
    })

    it('ProvideLastName_WhenInvalid_ShouldGenerateInvalidMessage', () => {
        const component = shallow(<Signup/>)
        component.instance().handleLName('kumar');
        component.instance().validateLName();
        expect(component.instance().state.lnameValid).toBe('invalid')
    })

    it('ProvideLastName_WhenValid_ErrorMessageShouldRemoved', () => {
        const component = shallow(<Signup/>)
        component.instance().handleLName('kumar');
        component.instance().validateLName();
        expect(component.instance().state.lnameValid).toBe('invalid');
        component.instance().handleLName('Kumar');
        component.instance().validateLName();
        expect(component.instance().state.lnameValid) .toBe ('');
    })

    it('ProvideUserName_WhenInvalidFormat_ShouldGenerateInvalidMessage', () => {
        const component = shallow(<Signup/>)
        component.instance().handleUserName('1234567');
        component.instance().validateUserName();
        expect(component.instance().state.userNameValid).toBe('Invalid Email')
    })

    it('ProvideUserName_WhenValid_ErrorMessageShouldRemove', () => {
        const component = shallow(<Signup/>)
        component.instance().handleUserName('1234567');
        component.instance().validateUserName();
        expect(component.instance().state.userNameValid).toBe('Invalid Email');
        component.instance().handleUserName('pk.soft29@gmail.com');
        component.instance().validateUserName();
        expect(component.instance().state.userNameValid).toBe('')
    })

    it('ProvidePassword_WhenInvalidFormat_ShouldGenerateInvalidMessage', () => {
        const component = shallow(<Signup/>)
        component.instance().handlePassword('1234567');
        component.instance().validatePassword();
        expect(component.instance().state.passwordValid).toBe('Not Strong')
    })

    it('ProvidePassword_WhenValid_ErrorMessageShouldRemoved', () => {
        const component = shallow(<Signup/>)
        component.instance().handlePassword('1234567');
        component.instance().validatePassword();
        expect(component.instance().state.passwordValid).toBe('Not Strong');
        component.instance().handlePassword('Pk@16123114');
        component.instance().validatePassword();
        expect(component.instance().state.passwordValid).toBe('');
    })

    it('ProvideConfirmPassword_WhenNotMatchecWithPassword_ShouldGenerateErrorMessage', () => {
        const component = shallow(<Signup/>)
        component.instance().handlePassword('pk.soft29');
        component.instance().handleConfirm('abc@12345');
        component.instance().comparePassword();
        expect(component.instance().state.passMatch).toBe('Wrong Password')
    })

    it('ProvideConfirmPassword_WhenMatchecWithPassword_ErrorMessageShouldRemoved', () => {
        const component = shallow(<Signup/>)
        component.instance().handlePassword('pk.soft29');
        component.instance().handleConfirm('abc@12345');
        component.instance().comparePassword();
        expect(component.instance().state.passMatch).toBe('Wrong Password');
        component.instance().handleConfirm('pk.soft29');
        component.instance().comparePassword();
        expect(component.instance().state.passMatch).toBe('');
    })
})
describe('Test Signup Button', () => {
    // it('PressSignupButton_WhenAllFieldValueIsCorrect_ShouldNavigateToLogin ', async() => {
    //     const navigation = { navigate: jest.fn()}
    //     const onPressEvent = jest.fn();
    //     const component = shallow(<Signup onPress = {onPressEvent} navigation = {navigation}/>);
    //     component.instance().handleFName('Pawan');
    //     component.instance().handleLName('Kumar');
    //     component.instance().handleUserName('pk.soft29@gmail.com');
    //     component.instance().handlePassword('PK@16123114');
    //     component.instance().handleConfirm('Pk@16123114')
        
    //     await component.instance().handleSignUpButton();

    //     expect(navigation.navigate).toBeCalledWith("Login");
    // })
    it('WhenPressSignupButton_ShouldNavigateToLogin ', async() => {
        const navigation = { navigate: jest.fn()}
        const onPressEvent = jest.fn();
        const component = shallow(<Signup onPress = {onPressEvent} navigation = {navigation}/>);
        
        await component.instance().SigninInsteadNavigationHandler();
        expect(navigation.navigate).toBeCalledWith("Login");
    })
})
    