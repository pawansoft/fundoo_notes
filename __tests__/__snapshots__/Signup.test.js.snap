// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Test ProvideRender_WhenMatchToSnapshot_TestShouldPass 1`] = `ShallowWrapper {}`;
